torch/csrc/jit/codegen/cuda/kernel_ir.cpp

```diff
@@ -269,7 +269,7 @@ TensorIndex::TensorIndex(
       "Cannot index with a value other than an int.");
 }
 
-Sync::Sync() : Expr(ExprType::Sync) {
+Sync::Sync(bool war_sync) : Expr(ExprType::Sync), war_sync_(war_sync) {
```

Comment:
This is a gray area: is this bit of information required for compiling and executing the kernel? If not, it should not be first class IR information.

We can, and should, discuss options for annotating IR with problem-specific information. For now, please add a TODO comment that the war_sync_ member is only used for testing/validation purposes.

Reply:
:+1: